.github/workflows/go.yml

```diff
@@ -33,3 +32,16 @@ jobs:
         env:
           COVERALLS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: goveralls -coverprofile=covprofile -service=github
+
+  staticcheck:
```

Comment:
Instead of using statticcheck, let's use golangci-lint (that i already pass locally when running the `make` command).

Very good idea to add this to the CI !

There are a bunch of other checks we do, look at the Makefile ;) Maybe we should do everything in the CI too !

Reply:
Okay, didn't realize this project had a Makefile.

Reply:
Went ahead and moved everything to work through golangci-lint. Updated makefile to provide targets that can help fix issues found in the linter.

Let me know if this is satisfactory to resolve.